### DISCUSSION TITLE
Job max attempts (cloud tasks vs $tries)

### DISCUSSION THREAD

**Discussion by mbreithu-wtd** [stackkit/laravel-google-cloud-tasks-queue#81] [Ideas]
Since V3.3 now ignores the backoff config from the task queue and uses the backoffs defined on the Laravel job, should the max retry count on the job also work like it does for normal laravel jobs? Currently it seems that the max retries value from the cloud tasks queue is used to determine a rescheduling of the job....

**Comment by marickvantuil** [comment]
I agree, that would make more sense. Might start working on that soon!

**Reply by mbreithu-wtd** [reply]
Maybe it would be possible to do it similar to regular laravel jobs/queues 

"One approach to specifying the maximum number of times a job may be attempted is via the --tries switch on the Artisan command line. This will apply to all jobs processed by the worker unless the job being processed specifies a more specific number of times it may be attempted"

So checking if a job specifies its own max retries and if not, fallback to the queue config (as it is currently)

**Reply by ttykhon** [reply]
I guess this wasn't ever implemented? Do you still have it planned?

**Reply by marickvantuil** [reply]
> I guess this wasn't ever implemented? Do you still have it planned?

Correct @ttykhon I never got around to changing this. It requires a major version because it's a breaking change. But since I want to drop some older Laravel versions, I might just make a `4.x` with this change in it.

**Reply by Plytas** [reply]
@marickvantuil sorry to ping you, but was this ultimately included in v4?

**Reply by marickvantuil** [reply]
@Plytas Yes, this was changed in 4.x - It now uses the job `$tries` instead of the Cloud Tasks queue configuration
